### PR TITLE
AC Automated Test - LRAC-12143, LRAC-12145 | master

### PR DESCRIPTION
**Pull request by manuferreira**
- Automation tickets:
https://issues.liferay.com/browse/LRAC-12143
https://issues.liferay.com/browse/LRAC-12145

- Testray run:
https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=2015187612&testrayRunId=2015191138

- `macro assertGenericResponse`
In this macro, you can pass whatever "path" to the response and assert the value.

- `macro getAnalyticsConfiguration`
This one is just the configuration we need in order to run the curl, so I just put everything inside a macro

- `macro getSearchHeaderValues`
In this case, I just created a static var (apparently in poshi this is the same as a global variable :shrug:) so this way is possible that we access the value of it inside another function like `macro getAnalyticsConfiguration`.

### CODE REVIEW

**Comment by manuferreira** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACApi.macro [hunk 1]
```diff
@@ -1,7 +1,31 @@
 definition {
 
+	macro assertGenericResponse {
```
[the remark this answers is not in the document]
Comment (manuferreira) [reply portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACApi.macro:3]: oh yes, I agree. Thank you, Josh.

**Comment by manuferreira** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase [hunk 2]
```diff
@@ -50,20 +47,43 @@ definition {
...
+	@description = "Feature ID: LRAC-12058 | Automation ID: LRAC-12143 | Test Summary: Fetch keywords with displayLanguageId parameter"
+	@priority = "3"
+	test RequestKeywordsUsingDisplayLanguageId {
+		property test.name.skip.portal.instance = "SearchAPI#RequestKeywordsUsingDisplayLanguageId";
+
+		var parametersDisplay1 = "size=10&displayLanguageId=${displayLanguageId1}&page=0&groupId=${groupId}&sort=counts,desc,lastModifiedDate,desc";
+		var parametersDisplay2 = "size=10&displayLanguageId=${displayLanguageId2}&page=0&groupId=${groupId}&sort=counts,desc,lastModifiedDate,desc";
+		var displayLanguageId1 = "en-US";
+		var displayLanguageId2 = "pt-BR";
+		var groupId = JSONGroupSetter.setGroupId(groupName = "Guest");
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase:82]: Hey @manuferreira, I think you don't need to use the `groupId` here. What do you think?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase [hunk 3]
```diff
@@ -143,6 +158,65 @@ definition {
...
 	}
 
+	@description = "Feature ID: LRAC-12058 | Automation ID: LRAC-12145 | Test Summary: Fetch keywords with page parameter"
+	@priority = "3"
+	test RequestKeywordsUsingPage {
+		property test.name.skip.portal.instance = "SearchAPI#RequestKeywordsUsingPage";
+
+		var parametersFirstPage = "size=${size}&displayLanguageId=en-US&page=${firstPage}&groupId=${groupId}&sort=counts,desc,lastModifiedDate,desc";
+		var parametersSecondPage = "size=${size}&displayLanguageId=en-US&page=${secondPage}&groupId=${groupId}&sort=counts,desc,lastModifiedDate,desc";
+		var groupId = JSONGroupSetter.setGroupId(groupName = "Guest");
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase:168]: Hey @manuferreira, I think you don't need to use the `groupId` here. What do you think?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase [hunk 4]
```diff
@@ -143,6 +158,65 @@ definition {
 		}
 	}
 
+	@description = "Feature ID: LRAC-12058 | Automation ID: LRAC-12145 | Test Summary: Fetch keywords with page parameter"
+	@priority = "3"
+	test RequestKeywordsUsingPage {
+		property test.name.skip.portal.instance = "SearchAPI#RequestKeywordsUsingPage";
+
+		var parametersFirstPage = "size=${size}&displayLanguageId=en-US&page=${firstPage}&groupId=${groupId}&sort=counts,desc,lastModifiedDate,desc";
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase:166]: Hey @manuferreira, I think you don't need to use the `displayLanguageId` here. What do you think?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase [hunk 5]
```diff
@@ -143,6 +158,65 @@ definition {
...
+		var parametersSecondPage = "size=${size}&displayLanguageId=en-US&page=${secondPage}&groupId=${groupId}&sort=counts,desc,lastModifiedDate,desc";
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase:167]: Hey @manuferreira, I think you don't need to use the `displayLanguageId` here. What do you think?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase [hunk 6]
```diff
@@ -143,6 +158,65 @@ definition {
...
+
+		task ("Assert response returns 0 for page") {
+			ACApi.assertGenericJSONPathResponse(
+				apiPath = "page.number",
+				expected = "${firstPage}");
+		}
+
+		task ("Assert response returns the second keyword") {
+			ACApi.assertSearchResponse(expected = "${secondKeyword}");
+		}
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase:203]: Hey @manuferreira, I think you can put another check here too to make sure the `${firstKeyword}` is not present in the second position of the response API (You need to pass `keywordPosition = "1"`)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase [hunk 7]
```diff
@@ -143,6 +158,65 @@ definition {
...
+
+		task ("Assert response returns 1 for page") {
+			ACApi.assertGenericJSONPathResponse(
+				apiPath = "page.number",
+				expected = "${secondPage}");
+		}
+
+		task ("Assert response returns first keyword") {
+			ACApi.assertSearchResponse(expected = "${firstKeyword}");
+		}
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase:217]: Hey @manuferreira, I think you can put another check here too to make sure the `${secondKeyword}` is not present in the second position of the response API (You need to pass `keywordPosition = "1"`). What do you think?

**Review by ViniLopes20 — APPROVED** [review]
Very good Manu, thanks for simplifying the test steps creating `getAnalyticsConfiguration`. I thought `assertGenericJSONPathResponse` was really cool and will be very reusable. Thank you for that 🎉 😄

**Review by joshitagaki — APPROVED** [review]
This is great! Thanks for improving and adding the macros to be easier to read for the testcase. LGTM!

**Review by calvinctvo — APPROVED** [review]
Hey @manuferreira ! Thanks for creating a macro for getting backend value and assert generic response! Great tests!

**Comment by calvinctvo** [comment]
Thanks guys for reviewing! I'm forwarding this PR

**Comment by calvinctvo** [comment]
ci:forward

**Pull request closed without merging**